gapic/templates/testing/constraints-3.7.txt.j2

```diff
@@ -1,12 +1,22 @@
+{# TODO(https://github.com/googleapis/gapic-generator-python/issues/2121): Remove the following variable (and the condition later in this file) for async rest transport once support for it is GA. #}
+{% set rest_async_io_enabled = api.all_library_settings[api.naming.proto_package].python_settings.experimental_features.rest_async_io_enabled %}
 {% from '_pypi_packages.j2' import pypi_packages %}
 # This constraints file is used to check that lower bounds
 # are correct in setup.py
 # List all library dependencies and extras in this file.
 # Pin the version to the lower bound.
 # e.g., if setup.py has "google-cloud-foo >= 1.14.0, < 2.0.0dev",
 # Then this file should have google-cloud-foo==1.14.0
+{% if rest_async_io_enabled %}
+google-api-core==2.20.0
+google-auth==2.35.0
+# from google-auth[aiohttp]
+aiohttp==3.6.2
+requests==2.20.0
```

Comment:
Consider creating a separate constraints file so that we know that the client library without the `async_rest` extra  works. See https://github.com/googleapis/python-api-core/pull/701 where we have a separate constraints file `testing/constraints-async-rest-3.7.txt` and separate nox session for testing `async_rest`

Reply:
addressed.